snapshot.go

```diff
@@ -228,6 +228,11 @@ func (r *Raft) compactLogs(snapIdx uint64) error {
 	// after the snapshot to be removed.
 	maxLog := min(snapIdx, lastLogIdx-r.conf.TrailingLogs)
 
+	if maxLog <= minLog {
```

Comment:
Should this be greater-than instead? @banks suggested "max is strictly larger than min". I lack some context but note that when I read "strictly larger" I wouldn't expect `<=`

Reply:
I would personally also do `minLog > maxLog` instead of `maxLog < minLog` for the conditional, I feel that would improve the readability. I had to read it 2x to realize what you had was correct, I think my mind just expected min/max and not max/min, but I admit that's my personal bias.

Reply:
Yeah the strictly part is necessar.

See the interface https://github.com/hashicorp/raft/blob/ff523e1c1515a5b33cc3153a34c984093fed06bb/log.go#L85 says the max an min are "inclusive" which means that min == max is actually valid and it will cause exactly one log entry to be deleted. This condition would consider `min:1000, max:1000` to be invalid and say "no logs to truncate" when in fact the current behaviour would be to truncate exactly one log with index 1000.

So yeah strict inequality is more correct here.